File: System/Protocol/Allors.Protocol.Json/Api/Push/PushRequestRole.cs
// <copyright file="PushRequestRole.cs" company="Allors bvba">
// Copyright (c) Allors bvba. All rights reserved.
// Licensed under the LGPL license. See LICENSE file in the project root for full license information.
// </copyright>

namespace Allors.Protocol.Json.Api.Push
{
    using System.Text.Json.Serialization;

    public class PushRequestRole
    {
        [JsonPropertyName("t")]
        public int RelationType { get; set; }

        [JsonPropertyName("u")]
        public string SetUnitRole { get; set; }

        [JsonPropertyName("c")]
        public long? SetCompositeRole { get; set; }

        [JsonPropertyName("a")]
        public long[] AddCompositesRole { get; set; }

        [JsonPropertyName("r")]
        public long[] RemoveCompositesRole { get; set; }
    }
}

File: System/Workspace/Typescript/libs/protocol/json/system/src/lib/api/Push/PushRequestRole.ts
export interface PushRequestRole {
  /** RelationType */
  t: number;

  /** SetUnitRole */
  u: string;

  /** SetCompositeRole */
  c: number;

  /** AddCompositesRole */
  a: number[];

  /** RemoveCompositesRole */
  r: number[];
}
